addons/payment/models/payment_acquirer.py

```diff
@@ -392,3 +392,27 @@ def _get_validation_currency(self):
     def _get_default_payment_method(self):
         self.ensure_one()
         return self.env.ref('account.account_payment_method_manual_in').id
+
+    def _neutralize_fields(self, provider, fields):
+        """ Helper to neutralize API keys for a specific provider
+            :param str provider: name of provider
+            :param list fields: list of fiels to nullify
+        """
+        acquirers = self.search([('provider', '=', provider)])
+        neutralized = {field_name: False for field_name in fields}
+        acquirers.write(neutralized)
+        for acquirer in acquirers:
+            _logger.info('%r acquirer %r account neutralized', acquirer.provider, acquirer.name)
+        not_neutralized_domain = [('provider', '=', provider)] + expression.OR([[(field_name, '!=', False)] for field_name in fields])
+        if self.search(not_neutralized_domain):
+            self._neutralize_warning("Not all acquirer account for `%s` were neutralized" % provider)
+
+    def _neutralize(self):
+        super()._neutralize()
+        acquirers = self.search([('state', '!=', 'disabled')])
+        acquirers.state = 'disabled'
+        for acquirer in acquirers:
+            _logger.info('Acquirer %r neutralized', acquirer.name)
+
+        if self.search([('state', 'not in', ['test', 'disabled'])]):
```

Comment:
```suggestion
        if self.search([('state', '!=', 'disabled')]):
```
to be coherent with the search above ;)